irohad/consensus/yac/impl/yac_gate_impl.cpp

```diff
@@ -42,28 +48,32 @@ namespace iroha {
             block_loader_(std::move(block_loader)),
             delay_(delay) {
         log_ = logger::log("YacGate");
-        block_creator_->on_block().subscribe([this](auto block) {
-          this->vote(*std::unique_ptr<iroha::model::Block>(block->makeOldModel()));
-        });
+        block_creator_->on_block().subscribe(
+            [this](auto block) { this->vote(*block); });
       }
 
-      void YacGateImpl::vote(model::Block block) {
-        auto hash = hash_provider_->makeHash(block);
+      void YacGateImpl::vote(const shared_model::interface::Block &block) {
+        std::unique_ptr<model::Block> bl(block.makeOldModel());
+        auto hash = hash_provider_->makeHash(*bl);
         log_->info(
-            "vote for block ({}, {})", hash.proposal_hash, hash.block_hash);
+            "vote for block ({}, {})", hash.proposal_hash, block.hash().toString());
```

Comment:
Please clarify is it possible for proposal_hash to differ from block_hash. It seems that it is redundant output and it is enough to print only `block.hash()`. If it is so, you don't need old model conversion at all.

Reply:
Also after that you can remove old block from the yac_gate_test.

Reply:
Will change it in the next PR